v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -250,21 +253,31 @@ func (r *AzureDeploymentReconciler) GetReconcilePolicy() ReconcilePolicy {
 	return policy
 }
 
-func (r *AzureDeploymentReconciler) makeReadyConditionFromError(cloudError *genericarmclient.CloudError) conditions.Condition {
+func (r *AzureDeploymentReconciler) makeReadyConditionFromError(cloudError *core.CloudError) (conditions.Condition, error) {
+	classifier := extensions.CreateErrorClassifier(r.extension, ClassifyCloudError, r.obj.GetAPIVersion(), r.log)
+	details, err := classifier(cloudError)
+	if err != nil {
+		return conditions.Condition{},
+			errors.Wrapf(
+				err,
+				"Unable to classify cloud error (%s)",
+				cloudError.Error())
+	}
+
 	var severity conditions.ConditionSeverity
-	errorDetails := ClassifyCloudError(cloudError)
-	switch errorDetails.Classification {
-	case CloudErrorRetryable:
+	switch details.Classification {
+	case core.ErrorRetryable:
 		severity = conditions.ConditionSeverityWarning
-	case CloudErrorFatal:
+	case core.ErrorFatal:
 		severity = conditions.ConditionSeverityError
 		// This case purposefully does nothing as the fatal provisioning state was already set above
 	default:
 		// TODO: Is panic OK here?
-		panic(fmt.Sprintf("Unknown error classification %q", errorDetails.Classification))
+		panic(fmt.Sprintf("Unknown error classification %q", details.Classification))
```

Comment:
(I think I was trying to avoid an `err` return here and this case is a bug in our code so panic felt ok-ish, but if we have an error return now anyway we might as well use it for this too)

Reply:
Done.